sdks/python/apache_beam/runners/worker/operation_specs.py

```diff
@@ -354,7 +354,10 @@ def get_coder_from_spec(coder_spec):
 
   # We pass coders in the form "<coder_name>$<pickled_data>" to make the job
   # description JSON more readable.
-  return coders.coders.deserialize_coder(coder_spec['@type'])
+  coder = coder_spec['@type']
+  if not isinstance(coder, bytes):
```

Comment:
Do we understand in which codepath  we happen to populate coders_spec without encoding it to bytes? If so, can we encode at the creation time? I think it would be easier to reason about SDK internals if we can state that this method always expects the same datatype (a bytestring) as input.